host/class/uvc/usb_host_uvc_2/private_include/uvc_types_priv.h

```diff
@@ -0,0 +1,58 @@
+/*
+ * SPDX-FileCopyrightText: 2024 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include <inttypes.h>
+#include <stdbool.h>
+#include <sys/queue.h>
+
+#include "usb/usb_host.h"
+#include "usb/uvc_host.h"
+
+#include "freertos/FreeRTOS.h"
+#include "freertos/queue.h"
+
+typedef struct uvc_host_stream_s uvc_stream_t;
+
+/**
+ * @brief Enum for simple state machine of Bulk frame data processing
+ */
+typedef enum {
+    UVC_STREAM_BULK_PACKET_SOF = 0,
+    UVC_STREAM_BULK_PACKET_DATA,
+    UVC_STREAM_BULK_PACKET_EOF,
+} uvc_stream_bulk_packet_type_t;
+
+struct uvc_host_stream_s {
```

Comment:
Once we agreed about the distinguish members by the way how they could be used (constant/dynamic/mux_protected). 

I still think that this is a good approach and it is better to keep using it. Especially in the new projects.

Reply:
Thanks for this comment, I refactored the structure as agreed before and it helped me find a few thread safety bugs.

The refactor is in a separate commit https://github.com/espressif/esp-usb/pull/84/commits/430f7d0124112ab93a8d773064942a66ad6155f0 for your convenience.

I've also added access to some dynamic members through `stdatomic.h` calls, to limit number of critical sections